Makefile

```diff
@@ -14,7 +14,15 @@
 
 MAKEFILE_PATH := $(abspath $(lastword $(MAKEFILE_LIST)))
 MAKEFILE_DIR  := $(dir $(MAKEFILE_PATH))
-NPROC ?= $(shell (nproc))
```

Comment:
as described in this thread https://github.com/4paradigm/OpenMLDB/pull/874#issuecomment-989707092, using the max cpu number may kill the build. We'd wondering if it is still necessary  to keep NPROC default to $(nproc)
@zhanghaohit

Reply:
Monitoring the CPU and memory usage during a compile on my VPS with 2 cores and 4GB memory.
### When ``` NPROC=1 ```
![Screen Shot 2021-12-10 at 12 35 38 PM](https://user-images.githubusercontent.com/1401515/145518735-f1836be1-9568-440e-8aa0-45dabcab1018.png)
![Screen Shot 2021-12-10 at 12 36 32 PM](https://user-images.githubusercontent.com/1401515/145518740-49371399-d5e7-4adc-a788-63a3152dd58c.png)
It takes nearly 2GB mem when compiling ``` avg ``` or ``` count ``` function.

### When ``` NPROC=2 ```
![Screen Shot 2021-12-10 at 12 38 15 PM](https://user-images.githubusercontent.com/1401515/145518996-eac94647-c0b4-4fef-b904-6aec2b931e10.png)
The screen shot shows the last output of ```top``` before the building process freezes the whole system. And we can see the cpu usage of ```[kswapd0]``` which indicates the whole system is running out of memory.

### Conclusion
So it is not how many building process that matters, even a VPS with 4cores and 4GB memory may be frozen by ```NPROC=2```.
**What should we do it to find out why compiling those two built-in function takes so much memory.**

Reply:
Nice experiment. Yet I don't know too much about the reason or how to improve it. Maybe with another build tool or dynamic link?
`avg_by_category_def.cc` compile is a known problem, I'll create a issue if not exists.

BTW can u try build with ninja instead of make, by `cmake ... -G Ninja`, it is said to have better memory usage.

Reply:
Tried with Ninja in my VPS, no luck when ```NPROC=2```